devito/ir/iet/scheduler.py

```diff
@@ -147,7 +147,7 @@ def iet_insert_decls(iet, external):
     allocator = Allocator()
     for k, v in MapSections().visit(iet).items():
         if k.is_Expression:
-            if k.is_scalar_assign:
+            if k.is_scalar_assign or k.is_tensor_assign:
```

Comment:
is it worth adding a ``is_definition`` property returning precisely this, that is ``self.is_scalar_assign or self.is_tensor_assign`` ?

Reply:
`is_scalar_assign` isn't used anywhere else so we can join the two together safely

Reply:
yep, I'd like that